### ISSUE TITLE
Datagrid: Does not stretch columns to cover the full width when it is initiated behind a tab

### ISSUE THREAD

**Issue by Rottosson** [infor-design/enterprise#2639]
**Describe the bug**
When a grids data, that is hidden behind a tab, is set it does not stretch columns to cover the full width.

**To Reproduce**
1. Run code below
2. Click on the Tab two
4. See that the columns are not stretched.
5. Click refresh to reset data and make the grid appear as intended
6. Move back to the first tab ( data has been reset there as well ) and the error should be shown again.

**Expected behavior**
Grids column should be able to handle data being set on grid that is hidden behind a tab.

**Version**
- ids-enterprise: 4.19.2
- ids-enterprise-ng: 5.4.0

**Platform**
 - Browser Name: chrome, internet explorer and edge.

**Additional context**
```
<button soho-button (click)="refresh()">refresh</button>

<div soho-tabs>
  <div soho-tab-list-container>
    <ul soho-tab-list>
      <li soho-tab><a soho-tab-title tabId="tab1">Tab one</a></li>
      <li soho-tab><a soho-tab-title tabId="tab2">Tab two</a></li>
    </ul>
  </div>
</div>

<div soho-tab-panel-container>
  <div soho-tab-panel tabId="tab1">
      <div
      soho-datagrid
      [data]="data"
      [gridOptions]="gridOptions">
    </div>
  </div>
  <div soho-tab-panel tabId="tab2">
    <div
      soho-datagrid
      [data]="data"
      [gridOptions]="gridOptions">
    </div>
  </div>
</div>


```

```
import { Component, OnInit } from '@angular/core';

@Component({
  templateUrl: './data-grid-demo-component.html',
})
export class DataGridDemoComponent implements OnInit {
  public data: any[] = [];
  readonly COLUMNS: SohoDataGridColumn[] = [
    { id: 'name', name: 'Name', field: 'name' },
    { id: 'time', name: 'Time', field: 'time' }
  ];
  public gridOptions: SohoDataGridOptions = {
    columns: this.COLUMNS,
    stretchColumn: 'last'
  };

  ngOnInit(): void {
    this.setData();
  }

  public setData(): void {
    this.data = [{
      name: 'Adam',
      time: '5'
    },
    {
      name: 'Bob',
      time: '3'
    }];
  }

  public refresh(): void {
    this.data = [];
    this.setData();
  }
}

```

**Comment by tmcconechy** [comment]
I found that this issue can more simply be reproduced in EP.

- edit components/datagrid/test-in-tabs.html so that instead of the third tab the first tab is selected (one without the datagrid on it)
- open the page
- click on the thrid tab with the datagrid 
- notice the last column is not stretched

```html
			<ul class="tab-list">
				<li class="tab is-selected"><a href="#tabs-normal-contracts">Contracts</a></li>
				<li class="tab"><a href="#tabs-normal-opportunities">Opportunities</a></li>
				<li class="tab"><a href="#tabs-normal-attachments">Attachments</a></li>
				<li class="tab"><a href="#tabs-normal-contacts">Contacts</a></li>
				<li class="tab"><a href="#tabs-normal-notes">Notes</a></li>
			</ul>
```

the problem is probably that the tab has no width when initialized so will need to add something to:

https://github.com/infor-design/enterprise/blob/master/src/components/datagrid/datagrid.js#L4202

To check if in a tab and grab a parent width.

**Comment by CarlPer** [comment]
@tmcconechy is there anything I can do until this is fixed?

**Comment by tmcconechy** [comment]
I will retest this after 4.24 (might fix it by fixing other issues). But i think what you could try as a workaround is. The assign each column a with in the column settings fx `width: 200` or whatever works for each column. Nothing will stretch but will maybe look ok.


Also you could try to use the afteractived event in tabs and invoke the datagrid at this point when the grid is visible. As the issue is it doesnt know how to size it correctly because the width is zero when invisible.

**Comment by tmcconechy** [comment]
Closing this issue as i finally had a chance to test this and it IS indeed fixed with the fixes we added in 4.24. If you retest on 4.24 with this steps: https://github.com/infor-design/enterprise/issues/2639#issuecomment-519925348 then the last column is now stretched out as its using a better table layout now.
